docs/payment-methods/FRENCH_MEAL_VOUCHER.md

```diff
@@ -0,0 +1,145 @@
+        val orderResponse = OrderResponse.SERIALIZER.deserialize(jsonResponse)
+        val dropInServiceResult = OrderDropInServiceResult.OrderCreated(orderResponse)
+        sendOrderResult(dropInServiceResult)
+    } else {
+        val dropInServiceResult = OrderDropInServiceResult.Error(..)
+        sendOrderResult(dropInServiceResult)
+    }
+}
+
+override fun onOrderCancel(order: Order, shouldUpdatePaymentMethods: Boolean) {
+    val orderJson = OrderRequest.SERIALIZER.serialize(order)
+    // Make a POST /orders/cancel request
+    if (isSuccessfulResponse()) {
+        if (shouldUpdatePaymentMethods) {
+            // shouldUpdatePaymentMethods is true when the shopper manually removes their meal vouchers and cancels the order
+            // The total reverts to the original amount and you might need to fetch your payment methods and update Drop-in with the new list of payment methods
+            val paymentMethods = fetchPaymentMethods() // Fetch the payment methods here
+            val dropInServiceResult = DropInServiceResult.Update(paymentMethods, null) // Update the payment methods
+        } else {
+            // shouldUpdatePaymentMethods is false when Drop-in is closed while the order is in progress
+            // If this happens, there is no need to make any further calls.
+        }
+    } else {
+        val dropInServiceResult = DropInServiceResult.Error(..)
+        sendResult(dropInServiceResult)
+    }
+}
+```
+
+### Configurations
+To create the configuration object and pass it when launching Drop-in, please follow the guide [here](https://docs.adyen.com/online-payments/build-your-integration/advanced-flow/?platform=Android&integration=Drop-in#configure).
+
+Additional configurations for French meal vouchers:
+```Kotlin
+CheckoutConfiguration(
+    environment = environment,
+    clientKey = clientKey,
+    ..
+) {
+    mealVoucherFR {
+        setSecurityCodeRequired(true)
+    }
+}
+```
+
+`setSecurityCodeRequired` allows you to specify if the Security code field should be hidden from the Component and not requested by the shopper. Note that this might have implications for the transaction.
+
+## Components
+Make sure to follow the [Android Components integration guide](https://docs.adyen.com/online-payments/build-your-integration/sessions-flow?platform=Android&integration=Components) and use the following module and component names:
+
+To [import the module](https://docs.adyen.com/online-payments/build-your-integration/sessions-flow/?platform=Android&integration=Components#import) use `meal-voucher-fr`.
+
+
+```Groovy
+implementation "com.adyen.checkout:meal-voucher-fr:YOUR_VERSION"
+```
+
+To [launch and show the Component](https://docs.adyen.com/online-payments/build-your-integration/sessions-flow?platform=Android&integration=Components#launch-and-show) use `MealVoucherFRComponent`.
+```Kotlin
+val component = MealVoucherFRComponent.PROVIDER.get(
+    activity = activity,
+    checkoutSession = checkoutSession,
```

Comment:
This code is for sessions, the main integration guide should be enough for sessions, this page should mainly be about the advanced flow IMO

Reply:
Maybe after mentioning the import and component class name, we can add a "sessions" section with just a link to the integration guide and an advanced section with all the details